sklearn/tests/test_isotonic.py

```diff
@@ -508,3 +509,36 @@ def test_make_unique_dtype():
         w = np.ones_like(x)
         x, y, w = _make_unique(x, y, w)
         assert_array_equal(x, [2, 3, 5])
+
+
+def test_input_shape_validation():
+    # Test from #15012
+    # Check that IsotonicRegression can handle 2darray with only 1 feature
+    X = np.arange(10)
+    X_2d = X.reshape(-1, 1)
+    y = np.arange(10)
+
+    iso_reg = IsotonicRegression().fit(X, y)
+    iso_reg_2d = IsotonicRegression().fit(X_2d, y)
+
+    y_pred1 = iso_reg.predict(X)
+    y_pred2 = iso_reg_2d.predict(X_2d)
+    assert_allclose(y_pred1, y_pred2)
```

Comment:
Please also check that the estimator attributes match in shape. Thanks

Reply:
Hi @jnothman, thanks for review. Are you referring to the `_necessary_X_` and `_necessary_y_` attributes? I have added checking for these two attributes. If it's something else, can you give a bit more details what you mean?

Reply:
Sorry, I meant the public attributes `X_min_`, `X_max_`, etc.

Reply:
hi @jnothman, sorry for the late reply. I have added the checks as requested. Could you review again?